core/storage/classifier/gae_models.py

```diff
@@ -181,6 +181,25 @@ def create(
         training_job_instance.put()
         return instance_id
 
+    @classmethod
+    def query_training_jobs(cls, inlcude_deleted=False):
+        """Generates a unique id for the training job of the form
+        {{exp_id}}.{{random_hash_of_16_chars}}
+
+        Args:
+            include_deleted: bool. If True, then entities that have been marked
+                deleted are returned as well. Defaults to False.
+        Returns:
+            List of the ClassifierTrainingJobModels with status new or pending.
+        """
+        query = cls.query(cls.status.IN([
+            feconf.TRAINING_JOB_STATUS_NEW,
+            feconf.TRAINING_JOB_STATUS_PENDING]))
+        if not inlcude_deleted:
+            query = query.filter(cls.deleted == False) # pylint: disable=singleton-comparison
+        query = query.order(cls.created_on, cls.id)
+        return query
```

Comment:
Just returning query does not give you models. You have to call `fetch` method on it to fetch jobs.

Also rather than fetching all jobs at once, fetch them page by page. That is fetch first 10 jobs and if you don't find a valid job in them then go and fetch next 10 (use the Cursor for that).